### ISSUE TITLE
ng build --prod stuck at 13% for long time (around 80%) of build time (total 30 mins to build)

### ISSUE THREAD

**Issue by mlakhara** [angular/angular-cli#12989]
### Bug Report or Feature Request (mark with an `x`)
```
- [ X] bug report -> please search issues before submitting
- [ ] feature request
```

### Command (mark with an `x`)
```
- [ ] new
- [ X] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Versions
Angular CLI: 7.0.3
Node: 10.13.0
OS: win32 x64
Angular: 7.0.1
... animations, common, compiler, compiler-cli, core, forms
... http, platform-browser, platform-browser-dynamic, router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.10.5
@angular-devkit/build-angular     0.10.5
@angular-devkit/build-optimizer   0.10.5
@angular-devkit/build-webpack     0.10.5
@angular-devkit/core              7.0.3
@angular-devkit/schematics        7.0.3
@angular/cli                      7.0.3
@ngtools/webpack                  7.0.5
@schematics/angular               7.0.3
@schematics/update                0.10.3
rxjs                              6.3.3
typescript                        3.1.4
webpack                           4.19.1

### Repro steps
Build Command:
`ng build -c=qa`

![stuck_at_13](https://user-images.githubusercontent.com/3102446/48700259-97317480-ec11-11e8-88e6-e26fe4ff6468.jpg)

As seen from the screenshot at this point it's only stuck at compiling the scss files I guess.

Build configuration
```
"optimization": true,
              "outputHashing": "all",
              "sourceMap": false,
              "extractCss": true,
              "namedChunks": false,
              "aot": true,
              "extractLicenses": true,
              "vendorChunk": false,
              "buildOptimizer": true
```

### Desired functionality
Any way this could be optimized. In previous version 1.7 this was not the case. Tried to use --profile=true but seems that is not working with current version.

**Comment by clydin** [comment]
Please update to the latest patch versions of the CLI (7.0.6), Angular (7.0.4), and TypeScript (3.1.6).

**Comment by mlakhara** [comment]
@clydin Thanks for the prompt response:

I updated the cli, angular and typescript to the latest version and the results are interesting.

The first time I ran the command, the build was stuck at 92% (at terser plugin) for a very long time after which I stopped that build, maybe it was because of node memory issue (although I had increased the limit to 10GB).

The second (and the third time), it worked
here is the breakdown

13% finished at 1m 50s
Module Concatenation plugin started around 4m17s and finished at 7m40s
And Terser Plugin started at 8m10s and finished at  9m10s

So that's great I guess, it brought down my build time from 30 mins (28 min with @angluar/cli@1.7 and @angular/core@5.1.2) to 9 minutes. So great efficiency on that too.

I tried to go through change log for cli and angular but couldn't find the exact change that resulted in this working. 

Anyways, thanks a lot.
